### PR TITLE
Popular articles within a topic: add parameter and poller

### PR DESCRIPTION
**Pull request by andygnewman**
@ironsidevsquincy @i-like-robots 

/articles endpoint take an optional parameter of topicid

When gets a new topicid, it adds it to the regular updater, though first search returns empty (should it return a diff status?)

topicids in updater would reset each time the app restarts.
Separate poller than standard most popular.

Keen queries are set off in max chunks of 5.

### CODE REVIEW

File: server/routes/articles.js [hunk 1]
```diff
@@ -3,24 +3,25 @@ import { logger } from 'ft-next-logger';
...
+						const errMsg = 'Error getting most popular popularArticles';
+						const status = 500;
+						logger.error(errMsg, err);
+						res
+								.status(status)
+								.json({
+										status: status,
+										message: `${errMsg}: ${err.message}`
+								});
+				});
```
Comment (i-like-robots) [inline server/routes/articles.js]: double tabs?

Comment (andygnewman) [inline server/routes/articles.js]: was quad spaced, but converted to tabs - will make it pretty.

**Comment by commuterjoy** [comment]
Yay, more people using Keen!

File: server/lib/popular-articles-topic-poller.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			}
+			if (this.updaterRunning) {
+				// run a one off update for this new topic
+				this.update([].concat(topicId), false);
+			} else {
+				// start the repeat updater
+				this.update();
+			}
+			// Don't wait for the result
+			return Promise.resolve(undefined);
```
Comment (ironsidevsquincy) [inline server/lib/popular-articles-topic-poller.js]: why not?

Comment (andygnewman) [inline server/lib/popular-articles-topic-poller.js]: I was thinking about this as well this morning as well. I originally had all this in stream app and I didn't want to slow down the process. However with having it here and with GraphQL intermediating then it's right to not wait. Will change it.

**Comment by ironsidevsquincy** [comment]
👍

**Pull request merged**